opre_ops/ops_site/migrations/0007_canamount_caninfo.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 3.2.6 on 2021-09-17 21:45
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('ops_site', '0006_alter_person_division'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='CANInfo',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('number', models.CharField(max_length=30)),
+                ('description', models.CharField(max_length=100)),
+                ('nickname', models.CharField(max_length=30)),
+                ('purpose', models.TextField(null=True)),
+                ('arrangement_type', models.CharField(choices=[('1', 'OPRE Appropriation'), ('2', 'Cost Share'), ('3', 'IAA'), ('4', 'IDDA'), ('5', 'MOU')], max_length=30)),
+                ('authorizer', models.ForeignKey(on_delete=django.db.models.deletion.PROTECT, related_name='authorizer', to='ops_site.agency')),
+                ('source', models.ManyToManyField(to='ops_site.Agency')),
+            ],
+            options={
+                'verbose_name_plural': 'CANs',
+            },
+        ),
+        migrations.CreateModel(
+            name='CANAmount',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('fiscal_year', models.CharField(max_length=5)),
```

Comment:
[_question_] The Google Sheet spec says DateField, not CharField. Which do we want to use here, and why? 

Are there advantages to CharField? If we go with CharField, should the max_length be 4 and not 5? 

My assumption is that using a DateField (or even an integer) would let us do math — e.g. `+= 1 year` operations — which we can't do on CharFields.

Reply:
Ah yes, I meant to ask the group about this. I'm not positive what we should do here. We'll never use a full date for just the year. If we need to do math on the year, I can possibly be convinced, but the DateField requires a month and day to be specified, which to me did not seem like something we would want or need. But, yes, open to discussion!

Reply:
Would be interested to hear @amymok's thoughts when she is back in the office because she authored the Google Sheet spec. 

I wonder if integer gives us the best of all worlds — you can get the next fiscal year given a current year if you need it (`next = current + 1`), no extraneous month or day data involved, some built-in validation:

```python
>>> int('garbage input')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'garbage input'
```

Reply:
@amymok do you have thoughts here?

Reply:
Oh oops, definitely missed this.  I chose DateField under the impression that it gives us more validation out of the box.  But since we are only using the year field, it may not be necessary because we probably don't need time delta or any of those.  So we can be flexible.  If we are giving the choices of CharField and IntegerField.  IntegerField would be better both for us to increment as well as to validate.

Here're some resources on how people have done it with choices:

- https://newbedev.com/year-field-in-django
- https://stackoverflow.com/questions/1517474/only-showing-year-in-django-admin-a-yearfield-instead-of-datefield/24656072#24656072

Reply:
Ok done! Changed to integer field/

Reply:
I actually ended up changing it to a DecimalField so that I could set a max_digits on it, max_length does not work on IntegerField.

Reply:
My first initial reaction was that DecimalField with `decimal_places=0` seems a little less idiomatic than an IntegerField for storing years. I pulled down this latest code using DecimalField and noticed that Django Admin formats the fiscal year correctly. No weird decimal places anywhere. I was able to increment the fiscal year seamlessly using the Django Admin UI. And I see how it gets us some easy-to-implement `max_length` validation @carjug. So I don't have any blocking suggestions or thoughts here! This isn't exactly what's in the Google Sheets spec but it doesn't seem unreasonable. Open to your input/feedback as one of the co-designers of the data model @amymok.

Reply:
The reason I used DecimaField instead of Integer is that IntegerField does not have a way to set a max_length. You can pass in `max_length` as a kwarg to `IntegerField` but it gets ignored. @alexsoble

Reply:
I would suggest we use `IntegerField` for a few reasons, (but also do not have the exact proofs though):
- it is more idiomatic to use `IntegerField`
- there may be unnecessary performance hit when we do comparison and filtering when using `DecimalField`
- we can use the `MidValueValidator` and `MaxValueValidator` to bound the field once we figure out what validation we want to do, I would not worry about needing it to have a `max_length` right now because there are still a big gap on what really is the validate numbers within that